### ISSUE TITLE
Pasting multiple clips pastes text but not images

### ISSUE THREAD

**Issue by sjman092** [sabrogden/Ditto#498]
I am trying to copy and paste both text and screenshot images that I have taken from a PDF file that I am making notes from.
When I try to select all clips and paste into a Word document, only the text is pasted.
Multiple images paste successfully into Word if I only select images but not if I select a mix of text and images.

**Comment by Chealer** [comment]
Thank you for reporting, but I am afraid you are misdiagnosing a considerably more complex bug.

It seems 03.25.113 can only reliably paste clips from one format at a time. If you try pasting clips from different formats, clips in a format other than the first clip's are typically silently ignored.

And when I say "format", I mean the custom clipboard formats like `CF_DIB` and `CF_TEXT`. So, even if I just try pasting multiple unformatted text clips, I get failures if the sets of formats of each clip differs. For example, if I copy "Foo" from the Windows Run dialog (which creates a `CF_TEXT`/`CF_UNICODETEXT` clip) and "Baz" from LibreOffice Writer (which creates a `CF_TEXT`/`CF_UNICODETEXT`/`HTML Format`/`Rich Text Format` clip), select both and try pasting, I get:

- Both lines in Notepad, as expected
- Both lines in Visual Studio Code, as expected
- Both lines in Microsoft Sticky Notes, as expected
- Both lines in GitHub Issues via Firefox, as expected
- Nothing but an "Erreur de lecture du presse-papier." error in Writer (that dialog is displayed by LibreOffice)
- Nothing in Google Docs via Firefox 146
- Only "Baz" and no error message in Google Docs via Google Chrome
- Only "Baz" and no error message in Thunderbird

But if "Baz" is first (that is, if it was copied before), then Writer pastes "Baz" (but still not "Foo").

To put it simply, it is just unreliable.

> Multiple images paste successfully into Word if I only select images but not if I select a mix of text and images.

As reported in ticket #60, even that is flaky.

Please retitle more accurately. I suggest:
> Trying to paste multiple clips in different formats silently ignores some (e.g. pastes text but not images, or vice-versa), or fails completely

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>
